api/client/builder/types.go

```diff
@@ -605,17 +563,28 @@ type BlobBundler interface {
 	BundleProto() (*v1.BlobsBundle, error)
 }
 
+// ParsedExecutionRequests can retrieve the underlying execution requests for the given execution payload response.
+type ParsedExecutionRequests interface {
+	ExecutionRequestsProto() (*v1.ExecutionRequests, error)
+}
+
 func (r *ExecutionPayloadResponse) ParsePayload() (ParsedPayload, error) {
 	var toProto ParsedPayload
-	switch r.Version {
-	case version.String(version.Bellatrix):
+	v, err := version.FromString(strings.ToLower(r.Version))
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("unsupported version %s", strings.ToLower(r.Version)))
+	}
+	if v >= version.Bellatrix {
 		toProto = &ExecutionPayload{}
-	case version.String(version.Capella):
+	}
+	if v >= version.Capella {
 		toProto = &ExecutionPayloadCapella{}
-	case version.String(version.Deneb):
+	}
+	if v >= version.Deneb {
 		toProto = &ExecutionPayloadDenebAndBlobsBundle{}
-	default:
-		return nil, consensusblocks.ErrUnsupportedVersion
+	}
+	if toProto == nil {
+		return nil, fmt.Errorf("unsupported version %s", strings.ToLower(r.Version))
 	}
```

Comment:
Doing it this way, you'll assign to `toProto` 3 times if the version is Deneb or later. Using `else` and reverting the order fixes it.
```
if v >= version.Deneb {
    toProto = &ExecutionPayloadDenebAndBlobsBundle{}
} else if v >= version.Capella {
    toProto = &ExecutionPayloadCapella{}
} else if v >= version.Bellatrix {
    toProto = &ExecutionPayload{}
} else {
    return nil, fmt.Errorf("unsupported version %s", strings.ToLower(r.Version))
}